pyproject.toml

```
@@ -44,6 +44,7 @@ classifier = [
 ]
 dependencies = [
     "pyarrow>=11.0.0",
+    "typing-extensions",
```

You probably want a python version bound on here. Like 
```suggestion
    "typing-extensions;python_version<'3.11'"
```
Do you know which version of Python is lacking the extended types?

---

It looks like you use typing.Self, which was [added in 3.11](https://docs.python.org/3/library/typing.html#typing.Self). It looks like typing.deprecated was [added in 3.13](https://peps.python.org/pep-0702/), so you probably want
```suggestion
    "typing-extensions;python_version<'3.13'",
```